packages/app/pages/index.tsx

```diff
@@ -360,6 +369,65 @@ const MembershipCard = (props: IMembershipCardProps) => {
               type={"number"}
             />
           </Modal>
+          <Modal
+            isOpen={airDropIsOpen}
+            setIsOpen={setAirDropIsOpen}
+            title="Airdrop Passports"
+            onConfirm={() => {
+              const contract = new web3.eth.Contract(
+                getAbiFromJson(passportJson)
+              );
+              contract.options.address = passport.address;
+              return Promise.all(airdropAddrList.map(addr => resolveAddress(addr, web3)))
+                .then(addrList => new Promise<void>((resolve, reject) =>
+                  contract.methods
+                  .airdrop(addrList, addrList.map((addr) => {return new BN(web3.utils.randomHex(32).replace(/^0x/, ""), "hex");}))
+                  .send({ 
+                    from: address
+                  })
+                  .then(() => {
+                    setToastMessage("Airdrop Successful!");
+                    setPassport({
+                      ...passport,
+                      supply: passport.supply-addrList.length,
+                    });
+                    setAirDropIsOpen(false);
+                    resolve();
+                  })
```

Comment:
we want this to `then` on receipt of the AirDrop event emitted instead of directly after send. The promise could resolve even if the transaction doesn't succeed/before it succeeds. Check out the buy or create Membership methods

Reply:
done